client/api/src/main/java/org/hawkular/apm/client/api/recorder/BatchTraceRecorder.java

```diff
@@ -139,7 +90,32 @@ public void run() {
                     }
                 }
             }
-        }, batchTime, batchTime, TimeUnit.MILLISECONDS);
+        }, builder.batchTime, builder.batchTime, TimeUnit.MILLISECONDS);
+
+        executor = Executors.newFixedThreadPool(builder.threadPoolSize,
+                new ThreadFactory() {
+                    public Thread newThread(Runnable r) {
+                        Thread t = Executors.defaultThreadFactory().newThread(r);
+                        t.setDaemon(true);
+                        return t;
+                    }
+                });
+    }
+
+    /**
+     * This method sets the trace publisher.
+     *
+     * @param tp The trace publisher
+     */
+    public void setTracePublisher(TracePublisher tp) {
```

Comment:
:+1: although It would be better to avoid adding getters/setters just for unit tests.